src/PwrDrvr.LambdaDispatch.Router/LambdaConnection.cs

```diff
@@ -377,7 +382,11 @@ private async Task RelayResponseFromLambda(HttpResponse incomingResponse)
       //
 
       // Read the status line
-      int endOfStatusLine = Array.IndexOf(headerBuffer, (byte)'\n');
+      // This buffer is huge and we may only have a small portion of it
+      // Do not read past the end of where we have written to
+      // The above loop will break out when ReadAsync returns 0,
+      // which can happen if a connection closes before the headers are finished
+      int endOfStatusLine = Array.IndexOf(headerBuffer, (byte)'\n', 0, totalBytesRead);
```

Comment:
This was previously reading previously used / un-initialized memory in the buffer because it was not limited `Array.IndexOf` to just the bytes that were written into the buffer.  If 0 bytes were read but the prior data (or random data) had a `\n` then this would think it found a status line and would then fail when trying to parse random data as the status code integer.